assets/js/util/scroll.js

```diff
@@ -65,6 +65,47 @@ export function getHeaderHeight( breakpoint ) {
 	return headerHeight;
 }
 
+/**
+ * Returns the height of the sticky WordPress admin bar, if present.
+ *
+ * @since n.e.x.t
+ *
+ * @param {string} breakpoint The current breakpoint.
+ * @return {number} The height of the sticky WordPress admin bar, if present.
+ */
+function getWordPressAdminBarHeight( breakpoint ) {
+	// WordPress admin bar is only sticky for breakpoints larger than BREAKPOINT_SMALL. If it's also not sticky then we can return a height of 0.
+	const wpAdminBar = document.querySelector( '#wpadminbar' );
+
+	if ( wpAdminBar && breakpoint !== BREAKPOINT_SMALL ) {
+		return wpAdminBar.offsetHeight;
+	}
+
+	return 0;
+}
+
+/**
+ * Returns the height of the sticky Site Kit header including the sticky WordPress admin bar when it's present.
+ *
+ * @since n.e.x.t
+ *
+ * @param {string} breakpoint The current breakpoint.
+ * @return {number} The height of the sticky Site Kit header including the sticky WordPress admin bar when it's present.
+ */
+function getGoogleSiteKitHeaderHeight( breakpoint ) {
+	// This function calculates the height of the sticky Site Kit header including the sticky WordPress admin bar when it's present.
+	const header = document.querySelector( '.googlesitekit-header' );
+	if ( header ) {
+		// If the breakpoint is BREAKPOINT_SMALL, the WordPress admin bar is not sticky and we can return the height of the Site Kit header.
+		// Otherwise, we use the value of the bottom of the Site Kit header's bounding box as this will take into account the sticky WordPress admin bar.
+		return breakpoint === BREAKPOINT_SMALL
+			? header.offsetHeight
+			: header.getBoundingClientRect().bottom;
```

Comment:
Hmm - the way this way originally written, there was a check for a negative result in the caller, to catch unexpected cases where the header is non-sticky. However, as you've pointed out we can make a more reliable check for stickiness in the caller, I think it would be a good idea to simply assume the header is sticky within this function - but, do what you've suggested just to handle the unexpected case here, i.e. do return `0` if `.bottom` is negative.